tests/test_integration/test_io.py

```diff
@@ -39,3 +41,64 @@ def test_convert_sleap_to_dlc_file(self, sleap_file, dlc_output_file):
         )
         dlc_ds = load_poses.from_dlc_file(dlc_output_file)
         xr.testing.assert_allclose(sleap_ds, dlc_ds)
+
+    @pytest.mark.parametrize(
+        "sleap_h5_file, fps",
+        [
+            ("SLEAP_single-mouse_EPM.analysis.h5", 30),
+            ("SLEAP_three-mice_Aeon_proofread.analysis.h5", None),
+            ("SLEAP_three-mice_Aeon_mixed-labels.analysis.h5", 50),
+        ],
+    )
+    def test_to_sleap_analysis_file_returns_same_h5_file_content(
+        self, sleap_h5_file, fps, new_h5_file
+    ):
+        """Test that saving pose tracks (loaded from a SLEAP analysis
+        file) to a SLEAP-style .h5 analysis file returns the same file
+        contents."""
+        sleap_h5_file_path = POSE_DATA.get(sleap_h5_file)
+        ds = load_poses.from_sleap_file(sleap_h5_file_path, fps=fps)
+        save_poses.to_sleap_analysis_file(ds, new_h5_file)
+
+        with h5py.File(ds.source_file, "r") as file_in, h5py.File(
+            new_h5_file, "r"
+        ) as file_out:
+            assert set(file_in.keys()) == set(file_out.keys())
+            keys = [
+                "track_occupancy",
+                "tracks",
+                "point_scores",
+            ]
+            for key in keys:
+                np.testing.assert_allclose(file_in[key][:], file_out[key][:])
+
+    @pytest.mark.parametrize(
+        "file",
+        [
+            "DLC_single-wasp.predictions.h5",
+            "DLC_single-wasp.predictions.csv",
+            "DLC_two-mice.predictions.csv",
+            "SLEAP_single-mouse_EPM.analysis.h5",
+            "SLEAP_single-mouse_EPM.predictions.slp",
+            "SLEAP_three-mice_Aeon_proofread.analysis.h5",
+            "SLEAP_three-mice_Aeon_proofread.predictions.slp",
+            "SLEAP_three-mice_Aeon_mixed-labels.analysis.h5",
+            "SLEAP_three-mice_Aeon_mixed-labels.predictions.slp",
+        ],
+    )
+    def test_to_sleap_analysis_file_source_file(self, file, new_h5_file):
```

Comment:
This test seems a bit over-parametrised to me.
Is it really necessary to test all of these cases? It doesn't hurt to be cautious I guess, but I'm a bit worried over the long-term impact on our test time (if we keep being so thorough).

Reply:
Thanks, good catch! I was just checking if it works in all cases. I've removed all but one test case per file type supported for each of DLC and SLEAP.